### PR TITLE
Incorporated axios + set up proxy

### PR DESCRIPTION
**Pull request by uglyprincess**
# Pull Request 

## Proposed Changes
- Rerouted data from signin + signup routes to backend
- Used axios for post requests, as well as http-proxy-middleware, and querystring
- Fixes Issue #27 

## Types of changes made
- [ ] Fixed Bug
- [x] Added new Features
- [x] Edited Documentations
- [x] Enhanced the working of code
- [ ] Additional Changes

## Additional Info
-

## Viability of Code
Describe the tests that you ran to verify the changes made. Mention the tests for easy reproducibility.
Tested the code on my local server.

#### Mention(*if any*) specific Test Configurations used:
- Filmware version:
- Hardware: 
- Toolchain: 
- SDK: 

## Screenshots or GIFs (*if any*)
Attach screenshots of the changes made for easy reference.
Original | Updated
:-----:|:-----:
*Attach_Original_Screenshot* | *Attach_Updated_Screenshot*

## Checklist:
- [x] My code follows the Mentioned Guidelines for the project
- [x] I have self-reviewed the changes made
- [x] I have made the changes easy to understand by adding comments
- [] My changes does not generate new warnings/errors
- [ ] I have attached relevant screenshots to display the changes made
- [x] I have made the required changes to the documentations

### CODE REVIEW

**Comment by uglyprincess** [comment]
@stygian-96 @Tlazypanda kindly review the changes made :))

**Comment by stygian-96** [comment]
@Tlazypanda please review.

File: src/components/auth-components/signin.js [hunk 1]
```diff
@@ -118,6 +125,39 @@ const useStyles = makeStyles((theme) => ({
...
+
+    const [info, updateInfo] = React.useState({
+        username: '',
+        password: ''
+    });
+
+    // This function will be called upon clicking the button and will send the login info to the back end.
+
+    function sendInfo(){
+        axios.post(`http://localhost:5000/login`, querystring.stringify({username: info.username, password: info.password}), {
```
Comment (Tlazypanda) [inline src/components/auth-components/signin.js]: Can we lose the localhost part here in this url?

File: src/components/auth-components/signin.js [hunk 2]
```diff
@@ -118,6 +125,39 @@ const useStyles = makeStyles((theme) => ({
...
+        }
+        console.log(response);
+      });
+    }
+
+    // Function to update the info.
+
+    const handleChange = (prop) => (event) => {
+        updateInfo({...info, [prop]: event.target.value});
+        console.log(info);
```
Comment (Tlazypanda) [inline src/components/auth-components/signin.js]: Log statements too if testing is done

File: src/components/auth-components/signin.js [hunk 3]
```diff
@@ -127,21 +167,31 @@ export default function CenteredGrid() {
...
+                                className={classes.inputText} 
+                                label="Username" 
+                                value={info.username}
+                                onChange={handleChange('username')}
+                                variant="outlined" />
+                                <TextField id="outlined-basic" 
+                                className={classes.inputText} 
+                                label="Password" 
+                                value={info.password}
+                                onChange={handleChange('password')}
```
Comment (Tlazypanda) [inline src/components/auth-components/signin.js:207]: Are validations added for password and cpassword

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (Tlazypanda) [inline .vscode/settings.json]: Some files here are not required remove them from commit

**Review by Tlazypanda — COMMENTED** [review]
@uglyprincess Good job! 🎉 💯  Have left some comments please address them and we are good to go!!

File: src/components/auth-components/signin.js (hunk 1 above)
Comment (uglyprincess) [reply src/components/auth-components/signin.js]: Will be resolved in latest commit!

File: src/components/auth-components/signin.js (hunk 2 above)
Comment (uglyprincess) [reply src/components/auth-components/signin.js]: Sure ma'am, I'll comment them out.

File: src/components/auth-components/signin.js (hunk 3 above)
Comment (uglyprincess) [reply src/components/auth-components/signin.js:207]: Validations for password will be added in the next commit.

File: .vscode/settings.json (hunk 4 above)
Comment (uglyprincess) [reply .vscode/settings.json]: Sure, will do!

**Comment by uglyprincess** [comment]
@Tlazypanda all the changes requested have been made. Please review! :))

**Review by Tlazypanda — COMMENTED** [review]
@uglyprincess still some files can be removed from commit like log ...once done merge this rest all looks good 👍

**Comment by uglyprincess** [comment]
Done ma'am! I believe those were the last of the useless files.

**Pull request merged**